### ISSUE TITLE
callCount not incremented if method is async and the ThrowableError is set (automockable)

### ISSUE THREAD

**Issue by otusweb** [krzysztofzablocki/Sourcery#1117]
when calling a mocked method for an async method, the callCount and arguments are not set if the throwable error is set

``` func saveTerminalSettings(_ settings: SwitchIOSettings) throws {
        if let error = saveTerminalSettingsThrowableError {
            throw error
        }
        saveTerminalSettingsCallsCount += 1
        saveTerminalSettingsReceivedSettings = settings
        saveTerminalSettingsReceivedInvocations.append(settings)
        try saveTerminalSettingsClosure?(settings)
    }
```

expected:
- saveTerminalSettingsCallsCount is incremented, receivedSettings is set, then the error is thrown

**Comment by rokridi** [comment]
This issue has been fixed in [version 2.1.3](https://github.com/krzysztofzablocki/Sourcery/releases#:~:text=%5B%231223%5D%20throw%20throwable%20error%20after%20updating%20calls%20count%20and%20received%E2%80%A6%20by%20%40rokridi%20in%20%231224)

**Comment by art-divin** [comment]
Currently generated code, as @rokridi commented, is following the OP requirements:
```swift
func callAsyncAndThrow(parameter: Int) async throws -> String {
        callAsyncAndThrowParameterIntStringCallsCount += 1
        callAsyncAndThrowParameterIntStringReceivedParameter = parameter
        callAsyncAndThrowParameterIntStringReceivedInvocations.append(parameter)
        if let error = callAsyncAndThrowParameterIntStringThrowableError {
            throw error
        }
        if let callAsyncAndThrowParameterIntStringClosure = callAsyncAndThrowParameterIntStringClosure {
            return try await callAsyncAndThrowParameterIntStringClosure(parameter)
        } else {
            return callAsyncAndThrowParameterIntStringReturnValue
        }
    }
```
